src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ApiController.cs

```diff
@@ -466,7 +466,7 @@ private string GetEncodedUrl(ShellSettings shellSettings, string token)
             var host = shellSettings.RequestUrlHosts.FirstOrDefault();
             var hostString = host != null ? new HostString(host) : Request.Host;
 
-            var pathString = HttpContext.Features.Get<ShellContextFeature>().OriginalPathBase;
+            var pathString = HttpContext.Features.Get<ShellContextFeature>()?.OriginalPathBase ?? PathString.Empty;
```

Comment:
Normally here we are sure that we are in a request scope with a non null httpContext `ShellContextFeature` but the additional null check is fine.